base/test/launcher/teamcity_service_messages.cc

```diff
@@ -0,0 +1,119 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/base/test/launcher/teamcity_service_messages.h"
+
+#include <string>
+
+#include "base/ranges/algorithm.h"
+#include "base/strings/string_number_conversions.h"
+
+namespace base {
+namespace {
+
+// https://www.jetbrains.com/help/teamcity/service-messages.html#Escaped+Values
+std::string_view TeamcityValueEscape(std::string_view s, std::string& result) {
```

Comment:
A combination of returning `string_view` and using `std::string& result` looks a little weird.
Could we just make it `WriteEscapedValueToStream(std::ostream &stream, std::string_view s)`?

Reply:
Reimplemented without string allocation.